pkgs/by-name/zo/zotify/package.nix

```
@@ -0,0 +1,58 @@
+{
+  fetchFromGitHub,
+  python3Packages,
+  lib,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "zotify";
+  version = "0.6.13";
+
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "zotify-dev";
+    repo = "zotify";
+    # repository has no version tags
+    rev = "5da27d32a1f522e80a3129c61f939b1934a0824a";
+    hash = "sha256-KA+Q4sk+riaFTybRQ3aO5lgPg4ECZE6G+By+x2uP/VM=";
+  };
+
+  build-system = [ python3Packages.setuptools ];
```

```suggestion
  build-system = [ python3.pkgs.setuptools ];
```

or `build-system = with python3.pkgs; [ setuptools ];`...more of a personal preference.

---

Unless `python3.pkgs` has been recently made to work with splicing, it isn't. The manual was specifically updated to advertise `python3Packages` instead of `python3.pkgs`: https://nixos.org/manual/nixpkgs/unstable/#buildpythonapplication-function

---

Sorry I wasn't more precise: https://github.com/NixOS/nixpkgs/commit/70aa345bec1135ea9524cd893aea1f6e3dd82387

---

Link to the actual issue: https://github.com/NixOS/nixpkgs/issues/211340.

This still is not really documented, though

---

I'm not sure which to use now. Most python packages I saw use `python3Packages`, so that's what I went with.

---

I'm pretty confident that in the current situation we should keep `python3Packages`

---

@fabaff ? @mweinelt ? @natsukium ? What's your POV on this?

---

python3Packages, for splicing reasons, as mentioned by SomeoneSerge.

---

For reference about this dreadful splicing issue, yay!, the issues are

#211340 (python but general)
#204303 (top-level)